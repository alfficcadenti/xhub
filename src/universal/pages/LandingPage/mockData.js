// export const CSRData = [
//     {brandName: 'BEX', CSRTrend: [1, 2, 3, 6, 7, 3, 4, 10], CSRByDevice: {mobile: 90, tablet: 99, desktop: 60}},
//     {brandName: 'Hotels.com', CSRTrend: [1, 2, 3, 6, 5, 11, 10, 6], CSRByDevice: {mobile: 10, tablet: 99, desktop: 60}},
//     {brandName: 'Vrbo', CSRTrend: [10, 9, 3, 4, 5, 6, 4, 9], CSRByDevice: {mobile: 97, tablet: 50, desktop: 60}},
// ];

export const CSRData = [
    {
        'time': '2020-05-21T10:30:00.000Z', BEX: 0.99, Vrbo: 0.1, 'Hotels.com': 0.1,
    },
    {
        'time': '2020-05-21T11:30:00.000Z', BEX: 0.9, Vrbo: 0.1, 'Hotels.com': 0.2,
    },
    {
        'time': '2020-05-21T12:30:00.000Z', BEX: 0.9, Vrbo: 0.1, 'Hotels.com': 0.3,
    },
    {
        'time': '2020-05-21T13:30:00.000Z', BEX: 0.9, Vrbo: 1, 'Hotels.com': 0.7,
    },
    {
        'time': '2020-05-21T14:30:00.000Z', BEX: 0.9, Vrbo: 0.9, 'Hotels.com': 0.98,
    },
    {
        'time': '2020-05-21T15:30:00.000Z', BEX: 0.5, Vrbo: 0.7, 'Hotels.com': 0.8,
    },
    {
        'time': '2020-05-21T16:30:00.000Z', BEX: 0.99, Vrbo: 0.89, 'Hotels.com': 0.72,
    },
];

export const bookingsData = [
    {
        'time': '2020-05-21T10:30:00.000Z', BEX: 4000, Vrbo: 2400, 'Hotels.com': 2400,
    },
    {
        'time': '2020-05-21T10:30:00.000Z', BEX: 3000, Vrbo: 1398, 'Hotels.com': 2210,
    },
    {
        'time': '2020-05-21T10:30:00.000Z', BEX: 2000, Vrbo: 9800, 'Hotels.com': 2290,
    },
    {
        'time': '2020-05-21T10:30:00.000Z', BEX: 2780, Vrbo: 3908, 'Hotels.com': 2000,
    },
    {
        'time': '2020-05-21T10:30:00.000Z', BEX: 1890, Vrbo: 4800, 'Hotels.com': 2181,
    },
    {
        'time': '2020-05-21T10:30:00.000Z', BEX: 2390, Vrbo: 3800, 'Hotels.com': 2500,
    },
    {
        'time': '2020-05-21T10:30:00.000Z', BEX: 3490, Vrbo: 4300, 'Hotels.com': 2100,
    },
];