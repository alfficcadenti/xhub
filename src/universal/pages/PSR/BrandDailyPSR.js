import React, {Component} from 'react';
import GaugeChart from '../../components/GaugeChart';
import h from './psrHelpers'
import PropTypes from 'prop-types';

class BrandDailyPSR extends Component {
    constructor(props) {
        super(props);
        this.state= {
            isOpen: false
        }
    }

    openDetails = () => {
        this.props.onClick(this.props.brand)
    }
    
    render() {
        const {
            brand,
            dailyPSRValue,
            date,
        } = this.props;

        return (
            <div className='brandPsr' key={brand + date} onClick={this.openDetails}>
            {
                h.brandLogoFile(brand) ? 
                <img className='brandLogoImg' alt={`${brand}-logo`} src={h.brandLogoFile(brand)} height="35"/> : 
                <h3 className='brandName'>{brand}</h3>
            }
            <div className='lastUpdate'>{date ? `Last update ${date}` : ''}</div>
            <GaugeChart title={brand} value={dailyPSRValue} />
        </div>
        );
    }
}

BrandDailyPSR.propTypes = {
    brand: PropTypes.string,
    dailyPSRValue : PropTypes.number,
    date: PropTypes.string,
    onClick: PropTypes.func
};
export default BrandDailyPSR;