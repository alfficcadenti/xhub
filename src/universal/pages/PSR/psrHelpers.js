/* eslint-disable complexity */
const distinct = (value, index, self) => {
    return self.indexOf(value) === index;
};

const listOfBrands = (data = []) => (data.map((x) => x.brand).filter(distinct));

const listOfLOB = (data = []) => (data.map((x) => x.lineofbusiness).filter(distinct));

const psrValuesByBrand = (data = [], brand) => (data.filter((x) => x.brand === brand));

const psrValuesByLOB = (data = [], lob) => (data.filter((x) => x.lineofbusiness === lob));

const psrValuesByDate = (data = [], lob) => (data.filter((x) => x.date === lob));

const findValuesByDate = (data = [], lob) => (data.find((x) => x.date === lob));

const psrValuesByInterval = (data = [], interval) => (data.filter((x) => x.interval === interval));

const findPSRValueByInterval = (data = [], interval) => (data.find((x) => x.interval === interval));

const psrDetailsByBrand = (data = [], brand) => (data.filter((x) => x.brand === brand));

const lastPSRAvailableDate = (data = []) => (data.reduce((acc, curr) => {
    return (curr.date > acc) ? curr.date : acc;
}, ''));

const getPSROnDate = (data = [], date = '') => (data.find((x) => x.date === date));

const formatDataForTable = (data = []) => (
    listOfLOB(data).map((lob) => {
        const daily = findPSRValueByInterval(psrValuesByLOB(data, lob), 'daily');
        const weekly = findPSRValueByInterval(psrValuesByLOB(data, lob), 'weekly');
        const monthly = findPSRValueByInterval(psrValuesByLOB(data, lob), 'monthly');
        return {
            'Line Of Business': lob,
            'Last 24 hours': (daily && `${daily.successPercentage.toFixed(2)} %`) || '',
            'Last 7 days': (weekly && `${weekly.successPercentage.toFixed(2)} %`) || '',
            'Last 28 days': (monthly && `${monthly.successPercentage.toFixed(2)} %`) || '',
        };
    })
);

const brandLogoFile = (brand) => {
    if (brand === 'vrbo' || brand === 'egencia' || brand === 'hcom') {
        try {
            return require(`../../img/logo-${brand}.png`).default;
        } catch {
            return null;
        }
    } else {
        return null;
    }
};

export default {
    listOfBrands,
    listOfLOB,
    psrValuesByBrand,
    psrValuesByLOB,
    psrValuesByDate,
    findValuesByDate,
    psrValuesByInterval,
    findPSRValueByInterval,
    psrDetailsByBrand,
    lastPSRAvailableDate,
    getPSROnDate,
    formatDataForTable,
    brandLogoFile
};
