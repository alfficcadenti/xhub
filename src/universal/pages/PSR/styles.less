@import "../../mixins.less";

div#psrContainer {
  text-align: center;

  div#dailyPsrContainer {
    display: flex;
    justify-content: center;

    div.brandPsr:hover {
      background-color: @hover-background;
      cursor: pointer;
    }
    div.brandPsr.selected {
      background-color: @selected-background;
      cursor: pointer;
    }

    div.brandPsr {
      float: left;
      text-align: center;
      padding-top: 10px;

      .brandName {
        font-size: 1.5em;
        height: 35px;
      }

      div.lastUpdate {
        font-weight: 300;
        font-style: oblique;
        margin-bottom: 1.2em;
        margin-top: 1.2em;
      }
    }
  }

  div#PSRDetails {
    width: 50%;
    justify-content: center;
    background-color: @selected-background;
    display: inline-block;

    table {
      width: 100%;
      white-space: pre-wrap;
      th {
        white-space: pre-wrap;
      }
    }

    th {
      text-align: center;
      background-color: @selected-background;
    }
  }
}
