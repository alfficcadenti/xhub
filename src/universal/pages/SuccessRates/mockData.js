export const successRatesData = [
    [{'time': '2020-08-03T11:20:00Z', 'successRatePercentagesData': [{'brand': 'All', 'rate': null}, {'brand': 'cheaptickets', 'rate': null}, {'brand': 'ebookers', 'rate': null}, {'brand': 'expedia', 'rate': null}, {'brand': 'hotwire', 'rate': null}, {'brand': 'lastminute', 'rate': null}, {'brand': 'nike', 'rate': null}, {'brand': 'orbitz', 'rate': null}, {'brand': 'travelocity', 'rate': null}, {'brand': 'vrbo', 'rate': null}, {'brand': 'wotif', 'rate': null}]}, {'time': '2020-08-03T11:25:00Z', 'successRatePercentagesData': [{'brand': 'All', 'rate': 86.70886075949366}, {'brand': 'cheaptickets', 'rate': null}, {'brand': 'ebookers', 'rate': 100}, {'brand': 'expedia', 'rate': 97.33333333333334}, {'brand': 'hotwire', 'rate': 100}, {'brand': 'lastminute', 'rate': 100}, {'brand': 'nike', 'rate': null}, {'brand': 'orbitz', 'rate': 100}, {'brand': 'travelocity', 'rate': 100}, {'brand': 'vrbo', 'rate': 73.13432835820896}, {'brand': 'wotif', 'rate': 75}]}, {'time': '2020-08-03T11:30:00Z', 'successRatePercentagesData': [{'brand': 'All', 'rate': 89.50617283950618}, {'brand': 'cheaptickets', 'rate': null}, {'brand': 'ebookers', 'rate': 100}, {'brand': 'expedia', 'rate': 98.78048780487805}, {'brand': 'hotwire', 'rate': null}, {'brand': 'lastminute', 'rate': 66.66666666666666}, {'brand': 'nike', 'rate': 100}, {'brand': 'orbitz', 'rate': 100}, {'brand': 'travelocity', 'rate': 100}, {'brand': 'vrbo', 'rate': 75}, {'brand': 'wotif', 'rate': 100}]}, {'time': '2020-08-03T11:35:00Z', 'successRatePercentagesData': [{'brand': 'All', 'rate': 84.57142857142857}, {'brand': 'cheaptickets', 'rate': 100}, {'brand': 'ebookers', 'rate': 100}, {'brand': 'expedia', 'rate': 98.91304347826086}, {'brand': 'hotwire', 'rate': null}, {'brand': 'lastminute', 'rate': null}, {'brand': 'nike', 'rate': 100}, {'brand': 'orbitz', 'rate': 100}, {'brand': 'travelocity', 'rate': 100}, {'brand': 'vrbo', 'rate': 62.857142857142854}, {'brand': 'wotif', 'rate': 100}]}, {'time': '2020-08-03T11:40:00Z', 'successRatePercentagesData': [{'brand': 'All', 'rate': 85.44303797468355}, {'brand': 'cheaptickets', 'rate': null}, {'brand': 'ebookers', 'rate': 100}, {'brand': 'expedia', 'rate': 94.68085106382979}, {'brand': 'hotwire', 'rate': 100}, {'brand': 'lastminute', 'rate': null}, {'brand': 'nike', 'rate': 100}, {'brand': 'orbitz', 'rate': null}, {'brand': 'travelocity', 'rate': null}, {'brand': 'vrbo', 'rate': 68.42105263157895}, {'brand': 'wotif', 'rate': 100}]}, {'time': '2020-08-03T11:45:00Z', 'successRatePercentagesData': [{'brand': 'All', 'rate': 88.30409356725146}, {'brand': 'cheaptickets', 'rate': null}, {'brand': 'ebookers', 'rate': 100}, {'brand': 'expedia', 'rate': 94.25287356321839}, {'brand': 'hotwire', 'rate': 100}, {'brand': 'lastminute', 'rate': 100}, {'brand': 'nike', 'rate': 100}, {'brand': 'orbitz', 'rate': 100}, {'brand': 'travelocity', 'rate': 100}, {'brand': 'vrbo', 'rate': 75}, {'brand': 'wotif', 'rate': 100}]}],
    [{'time': '2020-08-03T11:20:00Z', 'successRatePercentagesData': [{'brand': 'All', 'rate': null}, {'brand': 'cheaptickets', 'rate': null}, {'brand': 'ebookers', 'rate': null}, {'brand': 'expedia', 'rate': null}, {'brand': 'hotwire', 'rate': null}, {'brand': 'lastminute', 'rate': null}, {'brand': 'nike', 'rate': null}, {'brand': 'orbitz', 'rate': null}, {'brand': 'travelocity', 'rate': null}, {'brand': 'vrbo', 'rate': null}, {'brand': 'wotif', 'rate': null}]}, {'time': '2020-08-03T11:25:00Z', 'successRatePercentagesData': [{'brand': 'All', 'rate': 86.70886075949366}, {'brand': 'cheaptickets', 'rate': null}, {'brand': 'ebookers', 'rate': 100}, {'brand': 'expedia', 'rate': 97.33333333333334}, {'brand': 'hotwire', 'rate': 100}, {'brand': 'lastminute', 'rate': 100}, {'brand': 'nike', 'rate': null}, {'brand': 'orbitz', 'rate': 100}, {'brand': 'travelocity', 'rate': 100}, {'brand': 'vrbo', 'rate': 73.13432835820896}, {'brand': 'wotif', 'rate': 75}]}, {'time': '2020-08-03T11:30:00Z', 'successRatePercentagesData': [{'brand': 'All', 'rate': 89.50617283950618}, {'brand': 'cheaptickets', 'rate': null}, {'brand': 'ebookers', 'rate': 100}, {'brand': 'expedia', 'rate': 98.78048780487805}, {'brand': 'hotwire', 'rate': null}, {'brand': 'lastminute', 'rate': 66.66666666666666}, {'brand': 'nike', 'rate': 100}, {'brand': 'orbitz', 'rate': 100}, {'brand': 'travelocity', 'rate': 100}, {'brand': 'vrbo', 'rate': 75}, {'brand': 'wotif', 'rate': 100}]}, {'time': '2020-08-03T11:35:00Z', 'successRatePercentagesData': [{'brand': 'All', 'rate': 84.57142857142857}, {'brand': 'cheaptickets', 'rate': 100}, {'brand': 'ebookers', 'rate': 100}, {'brand': 'expedia', 'rate': 98.91304347826086}, {'brand': 'hotwire', 'rate': null}, {'brand': 'lastminute', 'rate': null}, {'brand': 'nike', 'rate': 100}, {'brand': 'orbitz', 'rate': 100}, {'brand': 'travelocity', 'rate': 100}, {'brand': 'vrbo', 'rate': 62.857142857142854}, {'brand': 'wotif', 'rate': 100}]}, {'time': '2020-08-03T11:40:00Z', 'successRatePercentagesData': [{'brand': 'All', 'rate': 85.44303797468355}, {'brand': 'cheaptickets', 'rate': null}, {'brand': 'ebookers', 'rate': 100}, {'brand': 'expedia', 'rate': 94.68085106382979}, {'brand': 'hotwire', 'rate': 100}, {'brand': 'lastminute', 'rate': null}, {'brand': 'nike', 'rate': 100}, {'brand': 'orbitz', 'rate': null}, {'brand': 'travelocity', 'rate': null}, {'brand': 'vrbo', 'rate': 68.42105263157895}, {'brand': 'wotif', 'rate': 100}]}, {'time': '2020-08-03T11:45:00Z', 'successRatePercentagesData': [{'brand': 'All', 'rate': 88.30409356725146}, {'brand': 'cheaptickets', 'rate': null}, {'brand': 'ebookers', 'rate': 100}, {'brand': 'expedia', 'rate': 94.25287356321839}, {'brand': 'hotwire', 'rate': 100}, {'brand': 'lastminute', 'rate': 100}, {'brand': 'nike', 'rate': 100}, {'brand': 'orbitz', 'rate': 100}, {'brand': 'travelocity', 'rate': 100}, {'brand': 'vrbo', 'rate': 75}, {'brand': 'wotif', 'rate': 100}]}],
    [{'time': '2020-08-03T11:20:00Z', 'successRatePercentagesData': [{'brand': 'All', 'rate': null}, {'brand': 'cheaptickets', 'rate': null}, {'brand': 'ebookers', 'rate': null}, {'brand': 'expedia', 'rate': null}, {'brand': 'hotwire', 'rate': null}, {'brand': 'lastminute', 'rate': null}, {'brand': 'nike', 'rate': null}, {'brand': 'orbitz', 'rate': null}, {'brand': 'travelocity', 'rate': null}, {'brand': 'vrbo', 'rate': null}, {'brand': 'wotif', 'rate': null}]}, {'time': '2020-08-03T11:25:00Z', 'successRatePercentagesData': [{'brand': 'All', 'rate': 86.70886075949366}, {'brand': 'cheaptickets', 'rate': null}, {'brand': 'ebookers', 'rate': 100}, {'brand': 'expedia', 'rate': 97.33333333333334}, {'brand': 'hotwire', 'rate': 100}, {'brand': 'lastminute', 'rate': 100}, {'brand': 'nike', 'rate': null}, {'brand': 'orbitz', 'rate': 100}, {'brand': 'travelocity', 'rate': 100}, {'brand': 'vrbo', 'rate': 73.13432835820896}, {'brand': 'wotif', 'rate': 75}]}, {'time': '2020-08-03T11:30:00Z', 'successRatePercentagesData': [{'brand': 'All', 'rate': 89.50617283950618}, {'brand': 'cheaptickets', 'rate': null}, {'brand': 'ebookers', 'rate': 100}, {'brand': 'expedia', 'rate': 98.78048780487805}, {'brand': 'hotwire', 'rate': null}, {'brand': 'lastminute', 'rate': 66.66666666666666}, {'brand': 'nike', 'rate': 100}, {'brand': 'orbitz', 'rate': 100}, {'brand': 'travelocity', 'rate': 100}, {'brand': 'vrbo', 'rate': 75}, {'brand': 'wotif', 'rate': 100}]}, {'time': '2020-08-03T11:35:00Z', 'successRatePercentagesData': [{'brand': 'All', 'rate': 84.57142857142857}, {'brand': 'cheaptickets', 'rate': 100}, {'brand': 'ebookers', 'rate': 100}, {'brand': 'expedia', 'rate': 98.91304347826086}, {'brand': 'hotwire', 'rate': null}, {'brand': 'lastminute', 'rate': null}, {'brand': 'nike', 'rate': 100}, {'brand': 'orbitz', 'rate': 100}, {'brand': 'travelocity', 'rate': 100}, {'brand': 'vrbo', 'rate': 62.857142857142854}, {'brand': 'wotif', 'rate': 100}]}, {'time': '2020-08-03T11:40:00Z', 'successRatePercentagesData': [{'brand': 'All', 'rate': 85.44303797468355}, {'brand': 'cheaptickets', 'rate': null}, {'brand': 'ebookers', 'rate': 100}, {'brand': 'expedia', 'rate': 94.68085106382979}, {'brand': 'hotwire', 'rate': 100}, {'brand': 'lastminute', 'rate': null}, {'brand': 'nike', 'rate': 100}, {'brand': 'orbitz', 'rate': null}, {'brand': 'travelocity', 'rate': null}, {'brand': 'vrbo', 'rate': 68.42105263157895}, {'brand': 'wotif', 'rate': 100}]}, {'time': '2020-08-03T11:45:00Z', 'successRatePercentagesData': [{'brand': 'All', 'rate': 88.30409356725146}, {'brand': 'cheaptickets', 'rate': null}, {'brand': 'ebookers', 'rate': 100}, {'brand': 'expedia', 'rate': 94.25287356321839}, {'brand': 'hotwire', 'rate': 100}, {'brand': 'lastminute', 'rate': 100}, {'brand': 'nike', 'rate': 100}, {'brand': 'orbitz', 'rate': 100}, {'brand': 'travelocity', 'rate': 100}, {'brand': 'vrbo', 'rate': 75}, {'brand': 'wotif', 'rate': 100}]}],
    [{'time': '2020-08-03T11:20:00Z', 'successRatePercentagesData': [{'brand': 'All', 'rate': null}, {'brand': 'cheaptickets', 'rate': null}, {'brand': 'ebookers', 'rate': null}, {'brand': 'expedia', 'rate': null}, {'brand': 'hotwire', 'rate': null}, {'brand': 'lastminute', 'rate': null}, {'brand': 'nike', 'rate': null}, {'brand': 'orbitz', 'rate': null}, {'brand': 'travelocity', 'rate': null}, {'brand': 'vrbo', 'rate': null}, {'brand': 'wotif', 'rate': null}]}, {'time': '2020-08-03T11:25:00Z', 'successRatePercentagesData': [{'brand': 'All', 'rate': 86.70886075949366}, {'brand': 'cheaptickets', 'rate': null}, {'brand': 'ebookers', 'rate': 100}, {'brand': 'expedia', 'rate': 97.33333333333334}, {'brand': 'hotwire', 'rate': 100}, {'brand': 'lastminute', 'rate': 100}, {'brand': 'nike', 'rate': null}, {'brand': 'orbitz', 'rate': 100}, {'brand': 'travelocity', 'rate': 100}, {'brand': 'vrbo', 'rate': 73.13432835820896}, {'brand': 'wotif', 'rate': 75}]}, {'time': '2020-08-03T11:30:00Z', 'successRatePercentagesData': [{'brand': 'All', 'rate': 89.50617283950618}, {'brand': 'cheaptickets', 'rate': null}, {'brand': 'ebookers', 'rate': 100}, {'brand': 'expedia', 'rate': 98.78048780487805}, {'brand': 'hotwire', 'rate': null}, {'brand': 'lastminute', 'rate': 66.66666666666666}, {'brand': 'nike', 'rate': 100}, {'brand': 'orbitz', 'rate': 100}, {'brand': 'travelocity', 'rate': 100}, {'brand': 'vrbo', 'rate': 75}, {'brand': 'wotif', 'rate': 100}]}, {'time': '2020-08-03T11:35:00Z', 'successRatePercentagesData': [{'brand': 'All', 'rate': 84.57142857142857}, {'brand': 'cheaptickets', 'rate': 100}, {'brand': 'ebookers', 'rate': 100}, {'brand': 'expedia', 'rate': 98.91304347826086}, {'brand': 'hotwire', 'rate': null}, {'brand': 'lastminute', 'rate': null}, {'brand': 'nike', 'rate': 100}, {'brand': 'orbitz', 'rate': 100}, {'brand': 'travelocity', 'rate': 100}, {'brand': 'vrbo', 'rate': 62.857142857142854}, {'brand': 'wotif', 'rate': 100}]}, {'time': '2020-08-03T11:40:00Z', 'successRatePercentagesData': [{'brand': 'All', 'rate': 85.44303797468355}, {'brand': 'cheaptickets', 'rate': null}, {'brand': 'ebookers', 'rate': 100}, {'brand': 'expedia', 'rate': 94.68085106382979}, {'brand': 'hotwire', 'rate': 100}, {'brand': 'lastminute', 'rate': null}, {'brand': 'nike', 'rate': 100}, {'brand': 'orbitz', 'rate': null}, {'brand': 'travelocity', 'rate': null}, {'brand': 'vrbo', 'rate': 68.42105263157895}, {'brand': 'wotif', 'rate': 100}]}, {'time': '2020-08-03T11:45:00Z', 'successRatePercentagesData': [{'brand': 'All', 'rate': 88.30409356725146}, {'brand': 'cheaptickets', 'rate': null}, {'brand': 'ebookers', 'rate': 100}, {'brand': 'expedia', 'rate': 94.25287356321839}, {'brand': 'hotwire', 'rate': 100}, {'brand': 'lastminute', 'rate': 100}, {'brand': 'nike', 'rate': 100}, {'brand': 'orbitz', 'rate': 100}, {'brand': 'travelocity', 'rate': 100}, {'brand': 'vrbo', 'rate': 75}, {'brand': 'wotif', 'rate': 100}]}]
];
