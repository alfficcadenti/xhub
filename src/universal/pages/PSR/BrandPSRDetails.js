import React, {PureComponent} from 'react';
import h from './psrHelpers'
import PropTypes from 'prop-types';
import DataTable from '../../components/DataTable/index';
import './styles.less';

class BrandPSRDetails extends PureComponent {
    render() {
        const {data} = this.props;
        const dataForTable = h.formatDataForTable(data);
        return (
            <div id='PSRDetails'>
                <DataTable
                    data={dataForTable}
                    columns={['Line Of Business','Last 24 hours','Last 7 days','Last 28 days']}
                    paginated={false}
                />
            </div>
        );
    }
}

BrandPSRDetails.propTypes = {
    data: PropTypes.array
};
export default BrandPSRDetails;