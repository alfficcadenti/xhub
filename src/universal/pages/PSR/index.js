/* eslint-disable complexity */
import React, {Component, Fragment} from 'react';
import 'moment-timezone';
import LoadingContainer from '../../components/LoadingContainer';
import BrandDailyPSR from './BrandDailyPSR';
import BrandPSRDetails from './BrandPSRDetails';
import h from './psrHelpers';
import './styles.less';

class PSR extends Component {
    constructor(props) {
        super(props);
        this.state = {
            data: [],
            isLoading: true,
            openDetails: false,
        };
    }

    componentDidMount() {
        this.loadPsr();
    }

    loadPsr = () => {
        fetch('/api/v1/psr')
            .then((resp) => {
                if (!resp.ok) {
                    this.setState({error: 'Psr not available. Try to refresh'});
                    throw new Error();
                }
                return resp.json();
            })
            .then((data) => {
                this.setState({
                    data,
                    isLoading: false
                });
            })
            .catch((error) => {
                // eslint-disable-next-line no-console
                console.log(error);
            });
    };

    lastDailyPSRValuesToDisplayByBrand = (data = []) => {
        const psrDailyData = h.psrValuesByLOB((h.psrValuesByInterval(data, 'daily')), 'PSR');
        return h.listOfBrands(psrDailyData)
            .map(
                (x) => {
                    const psrData = h.psrValuesByBrand(psrDailyData, x);
                    return h.getPSROnDate(psrData, h.lastPSRAvailableDate(psrData));
                }
            );
    };

    displayPSRDetails = (brand) => {
        this.setState({
            selectedBrand: brand,
            openDetails: true
        });
    };

    render() {
        const {isLoading, data, openDetails, selectedBrand} = this.state;
        const lastDailyPSRValuesByBrand = !isLoading && this.lastDailyPSRValuesToDisplayByBrand(data);
        const error = !isLoading && lastDailyPSRValuesByBrand.length === 0 ? 'Error: No Daily PSR to display' : '';
        const brandData = selectedBrand && h.psrDetailsByBrand(data, selectedBrand);

        return (
            <Fragment>
                <h1 className="page-title">{'Purchase Success Rates'}</h1>
                <LoadingContainer isLoading={isLoading} error={error}>
                    <div id="psrContainer">
                        <div id="dailyPsrContainer">
                            {
                                !isLoading && !error &&
                                lastDailyPSRValuesByBrand.map(
                                    (psr) => (
                                        <BrandDailyPSR
                                            key={`${psr.brand}PSRComponent`}
                                            brand={psr.brand}
                                            dailyPSRValue={psr.successPercentage}
                                            date={psr.date}
                                            onClick={this.displayPSRDetails}
                                            selected = {psr.brand === selectedBrand}
                                        />
                                    )
                                )
                            }
                        </div>
                        {
                            openDetails && selectedBrand && brandData.length !== 0 &&
                            <BrandPSRDetails
                                data={brandData}
                            />
                        }
                    </div>
                </LoadingContainer>
            </Fragment>
        );
    }
}

export default PSR;
