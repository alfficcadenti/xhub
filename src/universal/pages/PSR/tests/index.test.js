/* eslint-disable no-unused-vars */
import React from 'react';
import {expect} from 'chai';
import sinon from 'sinon';
import {shallow, mount} from 'enzyme';
import PSR from '../index';
import BrandDailyPSR from '../BrandDailyPSR'
import BrandPSRDetails from '../BrandPSRDetails'
import mockData from './data.test.json';

const mockDetails = [{
    "date": "2019-12-06",
    "brand": "egencia",
    "successPercentage": 84.0,
    "interval": "daily", 
    "lineofbusiness": "PSR"
  },
  {
    "date": "2019-12-06",
    "brand": "egencia",
    "successPercentage": 82.7,
    "interval": "monthly", 
    "lineofbusiness": "PSR Hotel"
  },
  {
    "date": "2019-12-06",
    "brand": "egencia",
    "successPercentage": 85.0,
    "interval": "weekly", 
    "lineofbusiness": "PSR Flights"
  }]

import {JSDOM} from 'jsdom';
const dom = new JSDOM('<!doctype html><html><body></body></html>');
global.window = dom.window;
global.document = dom.window.document;

describe('<PSR/>', () => {
    sinon.stub(PSR.prototype, 'componentDidMount');

    it('renders successfully', () => {
        const wrapper = shallow(<PSR />);
        expect(wrapper).to.have.length(1);
    });

    it('renders a div per brand', () => {
        const wrapper = mount(<PSR />);
        wrapper.setState({data: mockData, isLoading: false});
        expect(wrapper.find('div.brandPsr')).to.have.length(2);
    });
});

describe('<BrandDailyPSR/>', () => {
    const spy = sinon.spy();        
    const props = {
        key: 'vrboPSRComponent',
        brand: 'vrbo',
        dailyPSRValue: 98,
        date: '2019-01-01',
        onClick: spy
    };

    it('renders successfully', () => {
        const wrapper = shallow(<BrandDailyPSR />);
        expect(wrapper).to.have.length(1);
    });

    it('renders a div.brandPsr', () => {
        const wrapper = shallow(<BrandDailyPSR  {...props} />);
        wrapper.setState({isOpen: true});
        expect(wrapper.find('div.brandPsr')).to.have.length(1);
    });

    it('onClick trigger the func on props', () => {
        const wrapper = shallow(<BrandDailyPSR  {...props} />);
        wrapper.setState({isOpen: true});
        wrapper.find('div.brandPsr').simulate('click')
        expect(spy.calledOnce).to.be.true;
    });
});

describe('<BrandPSRDetails/>', () => {
    it('renders successfully the values for the different intervals', () => {
        const wrapper = shallow(<BrandPSRDetails data={mockDetails}/>);
        expect(wrapper.find('span#dailyPSR').text()).to.be.eql('84 %');
        expect(wrapper.find('span#weeklyPSR-Flights').text()).to.be.eql('85 %');
        expect(wrapper.find('span#monthlyPSR-Hotel').text()).to.be.eql('82.7 %');
    });
});