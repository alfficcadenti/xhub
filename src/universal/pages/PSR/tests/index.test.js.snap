// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
Array [
  <h1
    className="page-title"
  >
    Purchase Success Rates
  </h1>,
  <div
    className="loading-container"
    id={null}
  >
    <div
      className="LoadingOverlay fade in"
      role="status"
    >
      <ul
        aria-hidden={true}
        className="LoadingOverlay__ul"
      >
        <li
          className="LoadingOverlay__li LoadingOverlay__li--1 bg-brand"
        />
        <li
          className="LoadingOverlay__li LoadingOverlay__li--2 bg-brand"
        />
        <li
          className="LoadingOverlay__li LoadingOverlay__li--3 bg-brand"
        />
        <li
          className="LoadingOverlay__li LoadingOverlay__li--4 bg-brand"
        />
      </ul>
    </div>
  </div>,
]
`;

exports[`matches the snapshot 2`] = `
<div
  className="brandPsr selected"
  onClick={[Function]}
  onKeyDown={[Function]}
  role="button"
  tabIndex="0"
>
  <img
    alt="vrbo-logo"
    className="brandLogoImg"
    height="35"
    src={Object {}}
  />
  <div
    className="lastUpdate"
  >
    Last update 2019-01-01
  </div>
  <div
    className={undefined}
    id="vrbo"
    style={
      Object {
        "height": 150,
        "width": 250,
      }
    }
  />
</div>
`;
